app/search/models.py

```diff
@@ -56,7 +53,14 @@ def get_data(self):
         for result in results:
             data = result.__dict__
             data['descriptor'] = data['descriptor']
-            data.pop('_sa_instance_state', None)
             data['status'] = data['status'].value
+            p = result.publisher
+            data['publisher_name'] = p.name
+            if '_sa_instance_state' in data:
+                data.pop('_sa_instance_state', None)
```

Comment:
This becomes way too complicated than it needs to be. Can you just build the list of returned values like this?
```
for result in results:
    data_list.append({
         'status': result.status,
         'publisher_name': result.publisher.name,
         ... etc
    })
```